wrapper/CSharp/wolfSSL-TLS-Server/wolfSSL-TLS-Server.cs

```diff
@@ -50,16 +47,49 @@ private static void clean(IntPtr ssl, IntPtr ctx)
         wolfssl.Cleanup();
     }
 
+    /// <summary>
+    /// Checks if the SNI option was enabled via command line.
+    /// Must be enabled with ./configure --enable-sni when configuring
+    /// wolfSSL.
+    /// <param name="args">Parameters passed via command line</param>
+    /// </summary>
+    private static bool haveSNI(string[] args)
+    {
+        bool sniON = false;
+        for (int i = 0; i < args.Length; i++) {
+            if (args[i] == "-S") {
+                sniON = true;
+                break;
+            }
+        }
+        Console.WriteLine("SNI IS: " + sniON);
+        return sniON;
+    }
+
+    /// <summary>
+    /// Example of a SNI function call back
+    /// </summary>
+    /// <param name="ssl">pointer to ssl structure</param>
+    /// <param name="ret">alert code</param>
+    /// <param name="exArg">context arg, can be set with the function wolfssl.CTX_set_servername_arg</param>
+    /// <returns></returns>
+    public static int my_sni_server_cb(IntPtr ssl, IntPtr ret, IntPtr exArg) {
+        /* Trivial callback just for testing */
+        Console.WriteLine("my sni server callback");
+
+        return 0;
+    }
 
     public static void Main(string[] args)
     {
         IntPtr ctx;
         IntPtr ssl;
         Socket fd;
+        IntPtr arg_sni;
 
         /* These paths should be changed for use */
-        string fileCert = @"server-cert.pem";
-        string fileKey = @"server-key.pem";
+        string fileCert = @"../../certs/server-cert.pem";
+        string fileKey = @"../../certs/server-key.pem";

```

Comment:
this change has implications on the Windows VS build:

```
PS D:\work\sni-wrappers\wrapper\CSharp\DLL Debug\Win32> .\wolfSSL-TLS-Server.exe
Calling ctx Init from wolfSSL
Finished init of ctx .... now load in cert and key
Could not find cert or key file
freeing ctx handle
```

Reply:
Might consider logic like wolfssl/test.h ChangeToWolfRoot or gate the path based on `_WIN32`...